tonic/src/metadata/map.rs

```diff
@@ -198,11 +198,10 @@ pub struct OccupiedEntry<'a, VE: ValueEncoding> {
 
 impl MetadataMap {
     // Headers reserved by the gRPC protocol.
-    pub(crate) const GRPC_RESERVED_HEADERS: [&'static str; 8] = [
+    pub(crate) const GRPC_RESERVED_HEADERS: [&'static str; 7] = [
```

Comment:
So I think we should keep `grpc-timeout` here and instead expose a `Request::set_deadline` or something like that. What do you think?

Reply:
Afaik these headers are removed from all requests in `MetadataMap::into_sanitized_headers` so don't think that would work. In my testing that meant the `GrpcTimeout` middleware couldn't see the header value.

I do think adding `Request::set_deadline` is a good idea but given the odd formatting of the header value I don't think it'll be super easy. Think I'll look into that in a separate PR.

Reply:
Alright that is acceptable :D lets just get this shipped and we can focus on these things later.